<!DOCTYPE html>
<html>
  <head>
    <title>Developing a fine-mapping method for SNPs in high LD</title>
    <meta charset="utf-8">
    <meta name="author" content="Rene Welch" />
    <link rel="stylesheet" href="xaringan-themer.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Developing a fine-mapping method for SNPs in high LD
### Rene Welch
### 5-30-18

---









class: center, middle, inverse

# Introduction


---
class: center, middle
# Typical GWAS setting

&lt;img src="2018_05_30_finemapping_files/figure-html/ours/gwas1.png" width="500px" /&gt;

- If the coeff. are significant, we can say that a phenotype is associated to a region in the genome, because the genotypes are correlated

---
class: center, middle
## LD matrix example

&lt;img src="2018_05_30_finemapping_files/figure-html/LD_chr-1.png" width="500px" /&gt;

---
class: center, middle
## LD matrix example (part II)




&lt;img src="2018_05_30_finemapping_files/figure-html/artificial_manhattan-1.png" height="500" style="display: block; margin: auto;" /&gt;


---
class: left, top, inverse

# What is a fine-mapping method?

- Methods to assign well-calibrated probabilities of  causality to a set of candidate variants (Spain &amp; Barrett, 2015)

- Methods using genotype data:

  - Most associated SNP
  - Sequentially conditioning on SNPs.

- Methods using summary statistics:
$$
z | C = c \sim N( \Sigma \lambda_c , \Sigma )
$$
  - CAVIAR, (Hormozdiari et al, 2014 )
  
  - PAINTOR, prior over causal conf. depends on annotation data (Kichaev et al, 2014)
  
  - CAVIARBF, Bayesian fitting and numerical improvements (Chen et al, 2015 )
  
  - FINEMAP, Previous improvements + faster heuristic (Benner et al, 2016)

---
class: left, top

# Why do we need fine-maping methods?

- Adapting [Udler et al](https://onlinelibrary.wiley.com/doi/abs/10.1002/gepi.20504) to a continuous phenotype

&lt;img src="2018_05_30_finemapping_files/figure-html/power_calc-1.png" style="display: block; margin: auto;" /&gt;

- As the corr. increases, more individuals are required to recognize the causal SNP

- Those populations may be unavailable or even don't even exist

---
class: middle, center, inverse

# FM HighLD

---
class: left, top

# AROG framework

.pull-left[
&lt;img src="2018_05_30_finemapping_files/figure-html/ours/arog1.png" width="500px" /&gt;
]

.pull-right[

- We model the association statistic between a continuous phenotype and a SNP as in AROG (Shin and Keles, 2016)

- Which kind of annotation?

    - Peak overlaps
    - Sequence information (k-mer / motif scores)
    - atSNP scores
    - __Allelic skew__
]


&lt;!-- - For each gene, we build similar data structures as above and pool them together --&gt;

&lt;!-- - How do we extract loci where inflated effects are present? --&gt;


---
class: left, top

# FM HighLD

We minimize the following problem:
$$
f(C,\lambda) = \Vert C^T (z - A \lambda ) \Vert^2
$$

where:

- `\(A\)` is the annotation matrix.

- `\(z\)` is the vector of associations statistics between the SNPs and the phenotype.

### Solve the problem by iterating:

1. Given a causal conf. matrix `\(C\)`: `\(\mbox{min}_\lambda \Vert \tilde{z} - \tilde{A}\lambda \Vert^2\)`

2. Given a vector of annot. coeff. `\(\lambda\)`: `\(\mbox{min}_{C \in \mathcal{C} } \Vert C^T(z - A\lambda) \Vert^2\)`

---
class: top, left

# Fitting the model

The key step is to estimate the matrix:

`\begin{aligned}
C = \begin{bmatrix} C_1^T &amp; 0 &amp; 0 &amp; 0 &amp; \cdots &amp;0 \\ 0 &amp; C_2^T &amp; 0 &amp; 0 &amp; \cdots &amp; 0 \\ \vdots &amp; \vdots &amp; \vdots &amp; \ddots &amp; \cdots &amp; \vdots \\ 0 &amp; 0 &amp; 0 &amp; \cdots &amp; \cdots &amp; C_M^T\end{bmatrix}
\end{aligned}`

where `\(M\)` is the number of LD clusters in the loci.

## Our (baseline) approach:

For each LD cluster `\(i\)`: define a causal candidate as the SNP best represented by the annotation:

`\begin{aligned}
C_{ij}=\begin{cases}1 &amp;\text{if }j = \mbox{argmin}_k (z_{ik} - a_{ik}^T\hat{\lambda})^2\\ 0 &amp;  \text{otherwise} \end{cases}
\end{aligned}`

---
class: left, top

# EM algorithm

The main caveat of the previous approach, is that assumes 1 causal SNP for each LD cluster. Therefore, we relax that assumption by using a latent random variable:

`\begin{aligned}
W_i &amp;\sim \mbox{Ber}(p) \\
Z_i \mid A_i ,W_i = 0 &amp;\sim N(0,\sigma^2_0) \\
Z_i \mid A_i ,W_i = 1 &amp;\sim N(A_i^T \lambda,\sigma^2) \\
\end{aligned}`

### - E-step:

Calculate posterior probabilities for the causal states:

`\begin{aligned}
\hat{\gamma}_{i} = P(w_i = 1 \mid \tilde{z}_i , \tilde{A}_i,\lambda)
\end{aligned}`

### - M-step:

Fit a linear model, with weights `\(p_i = \hat{\gamma}_i\)` i.e. solve `\(\hat{\lambda} = \mbox{argmin} (\tilde{z} - \tilde{A}\lambda )^T \Gamma (\tilde{z} - \tilde{A}\lambda )\)`

---
class: left, top

# Full algorithm

0. Consider a set of regulatory SNPs in a loci(for example prioritize SNPs with one of the models stated in Kreimer et al 2017)

1. Cluster the SNPs based in their LD matrix and distance between them (for example we say two SNPs `\(s_i\)` and `\(s_j\)` in the same cluster if `\(\rho^2(s_i,s_j) \geq 0.75\)` and `\(d(s_i,s_j) \leq 1\)`MB)

3. Iterate the following steps (enhanced EM-algorithm):

    - For each LD cluster: select a causal candidate SNP
    - Estimate the probability of each LD cluster of containing a causal SNP
    - Fit a weighted linear model 

---
class: middle, center, inverse

# Annotation data

---
class: left, top

# Peak overlaps with Regulatory SNPs



&lt;img src="2018_05_30_finemapping_files/figure-html/enrichment_reg-1.png" style="display: block; margin: auto;" /&gt;

---
class: left, top

# Peak overlaps with expression modulating SNPs (cond. to Reg. SNPs)

&lt;img src="2018_05_30_finemapping_files/figure-html/enrichment_em-1.png" style="display: block; margin: auto;" /&gt;


---
class: left, middle

# Allelic skew pipeline

1. Mask SNP positions in the human genome (i.e. substitute the Ref / Alt bases for `N`)

2. Align FASTQ file to masked genome with `bowtie2`, and separate the alignment to each allele with [SNPsplit](https://www.bioinformatics.babraham.ac.uk/projects/SNPsplit/)

3. For every SNP:

      a. Count reads aligned to each allele
      
      b. For reads aligned to both alleles, solve the tie by randomly allocating them with prob 0.5
      
4. Calculate skew as:
$$
\mbox{skew} = \log_2( 1 + \mbox{alt}) - \log_2( 1 + \mbox{ref})
$$
5. To aggregate the reads (for example to make a TF-skew) do:

$$
\mbox{skew}_{\text{alle}}  = \log_2\left(1 + \sum_a \mbox{alt}_a 1(\mbox{alt}_a \geq \mbox{ref}_a) \right) - \log_2\left(  1 + \sum_a \mbox{ref}_a 1(\mbox{alt}_a &lt; \mbox{ref}_a)\right)
$$

---
class: left, top

# Exploration of allelic skew

&lt;img src="2018_05_30_finemapping_files/figure-html/asb_cover-1.png" width="800px" /&gt;

- Originally tried to use the enriched marks mentioned in [Tehwey et al](https://doi.org/10.1016/j.cell.2016.04.027) (mostly histone and DHS) 
- Got better results with ChIP-seq marks generated by [Reddy et al](https://genome.cshlp.org/content/22/5/860.long)
- In `*_alle` we apply the aggregation formula `\(\mbox{skew}_{\text{alle}}\)`

---
class: left, top

## Allele specific binding events

.pull-left[
### `TF` skew
&lt;img src="2018_05_30_finemapping_files/figure-html/asb_events-1.png" width="350px" /&gt;
]


.pull-right[
### `TF_alle` skew
&lt;img src="2018_05_30_finemapping_files/figure-html/asb_events_alle-1.png" width="350px" /&gt;

]

- By aggregating with the sum only, we were missing a lot of ASB events (defined by `p.adjust` `\(\leq\)` 0.05 in Binomial tests)

---
class: top,left

## Allelic skew models MPRA skew

&lt;img src="2018_05_30_finemapping_files/figure-html/asb_mpra-1.png" width="350px" style="display: block; margin: auto;" /&gt;

- __Quick note:__ Even though in the model, I used the aggregated ChIP-seq skew. 

      1. For the example loci, it is more clear with ATAC-seq
      2. For most of the loci, ATAC-skew and TF-skew are correlated

---
class: top,left

## Allelic skew models t-stat (almost)



.pull-left[

&lt;img src="2018_05_30_finemapping_files/figure-html/asb_atac-1.png" width="350px" style="display: block; margin: auto;" /&gt;

]

.pull-right[

&lt;img src="2018_05_30_finemapping_files/figure-html/asb_tf-1.png" width="350px" style="display: block; margin: auto;" /&gt;
]

| Gene | Pattern |
| :--- | :---------------------------------- |
| 177000 | Significant associations, similar values |
| 162441 | Decreasing trend between skew and tstat |



---
class: middle, center, inverse

# Continuous phenotype simulations

---
class: top, left

# Simulation framework

.pull-left[

We use a polygenic model:

`\begin{aligned}
y_i = \sum_{k = 1}^K x_{ik}\beta_k + \epsilon_i,\quad \epsilon_i\sim N(0,1-h^2_g)
\end{aligned}`

where

- `\(K\)` is the number of causal SNP
- `\(x_{ij}\)` are the genotype columns, centered and scaled
- `\(h_g^2\)` is the heritability

We calculate the t-statistics, by comparing the phenotype to every regulatory SNP one at a time:

`\begin{aligned}
y_i = x_{im}\beta^{(m)} + \varepsilon_i^{(m)},\quad\varepsilon_i^{(m)}\sim N(0,\tau^{(m)})
\end{aligned}`

]

.pull-right[

### Sampling strategies:

1. Select `\(K\)` random SNPs as causal

2. Sample `\(K\)` LD-clusters weighted by their average `\(R^2\)`, then select the causal SNP from the cluster randomly.


### Parameter definition

1. `\(\beta_k \propto \mbox{skew}_k\)`

2. Sample `\(\beta_k\)` from eQTL-SNP associations with `\(p_{val}\leq 10^{-64}\)`

]

---
class: top, left

# The SNP effect depends on the skew



&lt;img src="2018_05_30_finemapping_files/figure-html/sim1auroc-1.png" style="display: block; margin: auto;" /&gt;


---
class: top, left

# The SNP effect depends on the skew


&lt;img src="2018_05_30_finemapping_files/figure-html/sim1auprec-1.png" style="display: block; margin: auto;" /&gt;



---
class: top, left

# The SNP effect doesn't depend on the skew




&lt;img src="2018_05_30_finemapping_files/figure-html/sim2auroc-1.png" style="display: block; margin: auto;" /&gt;


---
class: top, left

# The SNP effect doesn't depend on the skew


&lt;img src="2018_05_30_finemapping_files/figure-html/sim2auprec-1.png" style="display: block; margin: auto;" /&gt;

---
class: middle, center, inverse

# Multi-phenotype analysis: 

# eQTL data

---
class: left, top

# eQTLs and MPRA

- eQTL studies work like GWAS, where the phenotype is gene expression 
- Several genes are tested, therefore we multiple gene - SNP combinations are tested
- In the loci below, there are 14 SNPs tested against 82 genes

&lt;img src="2018_05_30_finemapping_files/figure-html/mpra_eqtl-1.png" style="display: block; margin: auto;" /&gt;

---
class: left, top

.pull-left[
## # Genes tested per SNP
![](2018_05_30_finemapping_files/figure-html/gene_per_snp-1.png)&lt;!-- --&gt;


]


.pull-right[
## # SNPs tested per Gene


```
## # A tibble: 4 x 2
##   Restriction  `# of associations`
##   &lt;chr&gt;                      &lt;int&gt;
## 1 Total                        586
## 2 pval &lt;= 0.1                  100
## 3 pval &lt;= 0.05                  80
## 4 pval &lt;= 0.01                  52
```

- Not all the SNPs are tested against the same set of genes
- Majority of the associations are not significant

]

---
class: left

# Partitioning the data

&lt;img src="2018_05_30_finemapping_files/figure-html/gene_snp_partition-1.png" width="800px" /&gt;

- Create a graph via the relation: SNPs `a` and `b` are associated if they where tested for at least one common gene (used [`igraph`](http://igraph.org/r/) package), and partition the graph

- Originally used clusters with `\(\geq 10\)` SNPs 

- Used cluster with `\(\geq 5\)` SNPs, when pooling all the loci together

---
class: top, left

## Recall: Model for a single phenotype

.pull-left[

&lt;img src="2018_05_30_finemapping_files/figure-html/ours/scatter_skew_gene.png" width="450px" /&gt;

LD clusters: `\(k = 1,\cdots, n_g\)`

SNPs: `\(i =1,\cdots,n_{k}\)`

]

.pull-right[

Using a set of .important[regulatory] SNPs

1. Group the SNPs into LD clusters.

2. Given a set of causal candidates, iterate:

  i. Include a latent variable and calculate probability of each LD cluster of containing a causal SNP
  
  `\(w_{k} \sim \mbox{Ber}(\pi), \widetilde{z}_{k} \vert w_{k} = 1 \sim N( \widetilde{a}_{k}\lambda, \sigma^2_{k})\)`

  ii. Fit weighted .important[Linear model]:
      
  `\(\widetilde{z}_{k} = \widetilde{a}_{k} \lambda + \widetilde{\epsilon}_{k}\)`
        
  iii. Select another set of causal candidates 
]


---
class: top, left

## Model for multiple phenotypes

.pull-left[

&lt;img src="2018_05_30_finemapping_files/figure-html/ours/scatter_skew.png" width="450px" /&gt;

Genes: `\(g=1,\cdots,G\)`

LD clusters: `\(k = 1,\cdots, n_g\)`

SNPs: `\(i =1,\cdots,n_{gk}\)`

]

.pull-right[

Using a set of .important[regulatory] SNPs

1. Group the SNPs into LD clusters.

2. Given a set of causal candidates, iterate:

  i. Include a latent variable and calculate probability of each LD cluster of containing a causal SNP per gene
  
  `\(w_{gk} \sim \mbox{Ber}(\pi), \widetilde{z}_{gk} \vert w_{gk} = 1 \sim N( \widetilde{a}_{gk}\lambda, \sigma^2_{gk})\)`

  ii. Fit weighted .important[L. Mixed model]:
      
  `\(\widetilde{z}_{gk} = \widetilde{a}_{gk} \lambda + \widetilde{v}_{gk} u_g + \widetilde{\epsilon}_{gk}\)`
        
  iii. Select another set of causal candidates 
]

---
class: left, top

# Results of loci-specific models

Using allelic skew and ATAC-seq / ChIP-seq (histone) overlaps + slope and intercept by gene random effects




.pull-left[

- MPRA labels are unique per SNP, hence we summarized:
`\begin{aligned}
&amp;P(\text{SNP is causal}) = \\
&amp;\mbox{max}_g P(\text{SNP is causal for gene } g)
\end{aligned}`

- The peaks may be too wide, hence the annotation matrix may not be of full rank.

- Evaluating by loci is too restrictive, i.e. what is the min. number of SNPs to evaluate the model?
]

.pull-right[
&lt;img src="2018_05_30_finemapping_files/figure-html/loci_test-1.png" style="display: block; margin: auto;" /&gt;
]


---
class: left, top

# Pooling strategies



&lt;img src="2018_05_30_finemapping_files/figure-html/pooling_plot-1.png" style="display: block; margin: auto;" /&gt;

---
class: top, left

## Pooling strategies outperform ind. loci

&lt;img src="2018_05_30_finemapping_files/figure-html/comp1 -1.png" style="display: block; margin: auto;" /&gt;

---
class: top, left

## Shorter windows performs better 

&lt;img src="2018_05_30_finemapping_files/figure-html/tests_ov-1.png" style="display: block; margin: auto;" /&gt;


---
class: left, top

# Conclusions

.pull-left[

- FM HighLD outperforms other methods in simulated data

- FM HighLD is relatively fast, so we can evaluate the algorithm with multiple seeds and take the best 

- FM HighLD can evaluate multiple phenotype data, and estimate the probability of a SNP begin causal for multiple phenotypes

- Possible to extend this framework to colocalization problems

]

.pull-right[

## To do list:

- Adapt CAVIAR / PAINTOR to evaluate the eQTL data

- Measure the average run time

- Tehwey et al, provided a list of disease related loci. Match that list with the loci for which our model shows good performance

- Evaluate alternative annotation data: atSNP, conservation scores, suggestions please :) 

]



---
class: center, middle, inverse

# Thank you very much!

![](https://media.giphy.com/media/mjMMy4go77jiw/giphy.gif)


---
class: center, middle, inverse

# Extra slides

---
class: left, top

# Massive Parallel Reporter Assay (MPRA)

.pull-left[

&lt;img src="2018_05_30_finemapping_files/figure-html/from_papers/protocol2.png" width="250px" /&gt;

]


.pull-right[

Extracted from Tehwey et al., 2016

A. Oligos are synthesized as 180-mers around the variants

B. Single-stranded DNA is amplified, barcoded, and converted to double-stranded DNA by emulsion PCR, then cloned into a reporter vector

C. The plasmid library is linearized between the barcode and the oligos

D. The GFP is inserted to create the MPRA library

E. The library is transfected into the desired cell type

F. mRNA is captured and sequenced

]

---
class: left, top

# Skew imputation

`\begin{aligned}
\mbox{total skew} = \frac{\#(y_i^{\text{pred}}&gt;0)}{\# (y_i^{\text{obs}}&gt;0)} + \frac{\#(y_i^{\text{pred}}&lt;0)}{\#(:y_i^{\text{obs}}&lt;0)}
\end{aligned}`

&lt;img src="2018_05_30_finemapping_files/figure-html/load_imp -1.png" style="display: block; margin: auto;" /&gt;


---
class: left, top

# Derivation of CAVIAR framework

.pull-left[
Assume a GWAS model for 1 SNP:

`\begin{aligned}
y &amp;= \mu 1 + \beta_c x_c + e \\
\hat{\mu} = \bar{y} &amp;\quad \hat{\mu}\sim N(\mu,\sigma^2 / n) \\
\hat{\beta}_c = {x_c^Ty \over n } &amp;\quad \hat{\beta}_c \sim N(\beta_c,\sigma^2 / n) \\
\end{aligned}`

with `\(e \sim N(0,\sigma^2 I)\)` and `\(x_c\)` centered and scaled.

Then,

`\begin{aligned}
\hat{z}_c = { \sqrt{n}\beta_c / \sigma \over \sqrt{1/n} \sqrt{\hat{e}^T\hat{e}/\sigma^2}} \sim N(\lambda_c,1)
\end{aligned}`

with `\(\lambda_c = \beta_c\sqrt{n} / \sigma\)` and large `\(n\)`
]

.pull-right[

Assume other SNP `\(x_i\)` with `\({x_i^Tx_c\over n } = r\)`, then:

`\begin{aligned}
\hat{\beta_i} \sim N(r\beta_c ,\sigma^2 / n)
\end{aligned}`

which follows:

`\begin{aligned}
\begin{bmatrix}\hat{z}_c \\ \hat{z}_i \end{bmatrix} \sim N\left( \begin{bmatrix} 1 &amp; r \\ r &amp; 1 \end{bmatrix} \begin{bmatrix}\lambda_c \\ 0\end{bmatrix},\begin{bmatrix}1 &amp; r \\r &amp; 1 \end{bmatrix}\right)
\end{aligned}`

or in general, without knowledge of the causal SNPs in advance:

`\begin{aligned}
\hat{z}\mid C = c \sim N(\Sigma \lambda_c, \Sigma)
\end{aligned}`

where the entries of `\(\lambda_c\)` are the non-centrality parameters when causal and zero otherwise.

]
    </textarea>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function() {
  var d = document, s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})();</script>

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre']
  }
});
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
