<!DOCTYPE html>
<html>
  <head>
    <title>Rene Welch</title>
    <meta charset="utf-8">
    <link href="libs/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link href="libs/remark-css-0.0.1/default-fonts.css" rel="stylesheet" />
    <link rel="stylesheet" href="uw_slides.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Rene Welch
## Assistant scientist job talk
### 2018/04/17

---

class: top



## Rene Welch, PhD candidate in Statistics

.left-column2[
&lt;img src="figs/color-center-UWlogo-print.png" width="250px" /&gt;
]

.right-column2[

- Projects:

    - ChIPexoQual: A quality control pipeline for ChIP-exo/nexus data
    
    - Fine-mapping method for SNPs in High LD, associated to multiple responses
    
    - Biological effects of EBV (collaboration with Johannsen Lab)
    
    - Enhancer prediction (as part of ENCODE AWG)
    
- Analysis experience:

    - NGS: ChIP-seq/exo/nexus, DNase/ATAC-seq, RNA-seq, MeDIP-seq
    
    - Assays: MPRA, GWAS, eQTL
    
]

---
class: center, middle, inverse

# ChIPexoQual: A quality control pipeline for ChIP-exo/nexus data

---
class: center, middle

# ChIP-exo vs. ChIP-seq: Higher resolution

&lt;div class="figure"&gt;
&lt;img src="figs/rhee.png" alt="Coverage of different ChIP-protocols, Pugh &amp;amp; Rhee 2012" width="600px" /&gt;
&lt;p class="caption"&gt;Coverage of different ChIP-protocols, Pugh &amp; Rhee 2012&lt;/p&gt;
&lt;/div&gt;

---
class: middle

# ChIP-exo/nexus protocols

.pull-left[
&lt;img src="figs/chip_explanation_withEnzyme_nexus.png" width="500px" /&gt;
]

.pull-right[

# 

After DNA sonication:

- In ChIP-exo a `\(\lambda\)`-enzyme digests from the 5' end until it finds the TF, then one adapter is added for each side

- In ChIP-nexus, only one adapter is added at the 3' end, transferred to the 5' end by self-circularization. Unique fragments are distinguished by using random barcodes.

- In SE (PE) ChIP-seq, one (two) adaptors are added.
]

---
layout: true

background-image: url(figs/fastqc.png)
background-size: 600px
dpi: 10000

---
class: top, center 

## FASTQC: kmer-representation in ChIP-exo reads

---
class: top, center 

## FASTQC: kmer-representation in ChIP-exo reads

.pull-right[

### 

&lt;img src="figs/motif.png" width="300px" /&gt;

### FoxA1 motif

]

---
layout: false
class: top, justify

# PCR Bottleneck Coefficient

`\begin{aligned}
\mbox{PBC} = {\text{# of locs. with exactly 1 read}\over\text{# of locs. with }\geq 1\text{ reads}}
\end{aligned}`


&lt;img src="figs/PBC.png" width="500px" style="display: block; margin: auto;" /&gt;

- A experiment with .important[high library complexity] exhibits values `\(\approx 1\)`

- The majority of .important[ChIP-exo/nexus] samples are in the .important[0 - 0.5] range

---
class: top, justify

# Strand Cross-Correlation and related

`\begin{aligned}
\mbox{SCC}(\delta) = \sum_c w_c \rho(\mbox{fwd}_c - \delta , \mbox{bwd}_c + \delta)
\end{aligned}`

&lt;img src="figs/scc.png" width="700px" /&gt;

`\begin{aligned}
\mbox{NSC} = {\max_\delta \mbox{SCC}(\delta) \over \min_\delta \mbox{SCC}(\delta)  }, \quad
\mbox{RSC} = {\max_\delta \mbox{SCC}(\delta) - C \over \min_\delta \mbox{SCC}(\delta) - C  }
\end{aligned}`

- In .important[ChIP-exo]: both local maxima are closer together, could be confounded.

---
class: middle, center


### Available in the ChIPUtils R package

&lt;img src="figs/chiputils_me.png" width="700px" /&gt;

- .important[https://github.com/welch16/ChIPUtils/tree/devel]

---
class: top

# ChIPexoQual pipeline

Assuming a set of .important[N aligned reads]

1. Partition the genome and generate ChIP-exo islands

&lt;img src="figs/qc_pipeline_diagram.png" width="900px" /&gt;

---
class: top
# ChIPexoQual pipeline 
Assuming a set of .important[N aligned reads]

1. Partition the genome and generate ChIP-exo islands

2. Calculate summary statistics for each islands

`\begin{aligned}
D_i &amp;= \text{# of aligned reads in island } i \\
U_i &amp;= \text{# of unique positions that reads map in island } i \\
W_i &amp;= \text{width of island } i \\
\mbox{ARC}_i &amp;= D_i / W_i (\text{ represents sample enrichment}) \\
\mbox{URC}_i &amp;= U_i / D_i (\text{ represents library complexity}) 
\end{aligned}`

---
class: top
# ChIPexoQual pipeline 
Assuming a set of .important[N aligned reads]

1. Partition the genome and generate ChIP-exo islands

2. Calculate summary statistics for each islands

3. Visualize all islands together

&lt;img src="figs/qc_viz.png" width="700px" style="display: block; margin: auto;" /&gt;

---
class: top
# ChIPexoQual pipeline 
Assuming a set of .important[N aligned reads]

1. Partition the genome and generate ChIP-exo islands

2. Calculate summary statistics for each islands

3. Visualize all islands together

4. Calculate quality scores for the .important[ARC vs URC] plot:

      Re-parametrize:
`\begin{aligned}
\mbox{URC}_i = {\kappa \over \mbox{ARC}_i} + \gamma + \varepsilon'_i
\end{aligned}`

      into:
`\begin{aligned}
  D_i &amp;= \beta_1 U_i + \beta_2 W_i + \varepsilon_i ,\quad \varepsilon_i \sim N(0,\sigma^2) 
\end{aligned}`

---
class: top

# ChIPexoQual pipeline 

Assuming a set of .important[N aligned reads]

1. Partition the genome and generate ChIP-exo islands

2. Calculate summary statistics for each islands

3. Visualize all islands together

4. Calculate quality scores for the .important[ARC vs URC] plot:

5. If necessary, use extra modules: 

      - Subsampling approach for deeply sequenced samples.
      
      - Analysis of blacklisted regions.
      
      - Strand-imbalance analysis.


---
class: middle, center

### Available in the ChIPexoQual R package

&lt;img src="figs/bioconductor_me.png" width="500px" /&gt;

.important[http://www.bioconductor.org/packages/release/bioc/html/ChIPexoQual.html]


---
class: middle, center

# Balance between Average Read (ARC) and Unique Read (URC) Coefficients

1. Two arms: One with a decreasing trend and other with low .important[ARC] 

2. As .important[URC] decreases, the number of unique positions per region decreases

3. The regions with low .important[ARC], correspond to scattered reads in the genome

.pull-left[
&lt;img src="./figs/FoxA1_ARCvURC.gif" width="280px" /&gt;
]
.pull-right[
&lt;img src="./figs/FoxA1_profile.gif" width="400px" /&gt;
]

---
class: top, center

## Comparison between ChIP-exo/nexus TBP Quality Scores

.important[Recall:] We sample 1K regions, and fit the linear model (&gt;= 1K times):

`\begin{aligned}
  D_i &amp;= \beta_1 U_i + \beta_2 W_i + \varepsilon_i ,\quad \varepsilon_i \sim N(0,\sigma^2) ,
\end{aligned}`

.pull-left[
&lt;img src="figs/TBP_beta1.png" width="400px" /&gt;
]

.pull-right[
&lt;img src="figs/TBP_beta2.png" width="400px" /&gt;
]


---
class: top

## TBP ChIP-exo vs. ChIP-nexus at fixed sequencing depth


.pull-left[
&lt;img src="figs/TBP_beta1_samp.png" width="400px" /&gt;
]

.pull-right[
&lt;img src="figs/TBP_beta2_samp.png" width="400px" /&gt;
]

At fixed sequencing depth, ChIP-nexus is comparable (or even better) at fixed-sequencing depth.


---
class: inverse, middle, center

# Fine-mapping of SNPs in High LD associated to multiple eQTLs

---
class: top, center

## Motivation: Fine-mapping of SNPs in High LD

&lt;img src="./figs/finemap.png" width="660px" /&gt;


---
class: top, left

## A different annotation: Allelic skew

1. Mask SNP positions in human genome.

2. Align .important[assay reads] to masked genome with `bowtie2`, and allocated reads to each allele with [SNPsplit](https://www.bioinformatics.babraham.ac.uk/projects/SNPsplit/).

3. For every assay, calculate its allelic skew as:
`\begin{aligned}
\mbox{skew} = \log_2( 1 + \mbox{alt}) - \log_2( 1 + \mbox{ref})
\end{aligned}`
4. Aggregate the allele-aligned reads as .important[allelic skew:]

`\begin{align}
\text{aggregated skew}  = \log_2\left(1 + \sum_a \mbox{alt}_a 1(\mbox{alt}_a \geq \mbox{ref}_a) \right) - \log_2\left(  1 + \sum_a \mbox{ref}_a 1(\mbox{alt}_a &lt; \mbox{ref}_a)\right)
\end{align}`

---
class: top, left

## Description of the model for a single phenotype

.pull-left[

&lt;img src="figs/scatter_skew_gene.png" width="450px" /&gt;

LD blocks: `\(k = 1,\cdots, n_g\)`

SNPs: `\(i =1,\cdots,n_{k}\)`

]

.pull-right[

Using a set of .important[regulatory] SNPs

1. Group the SNPs into LD blocks.

2. Given a set of causal candidates, iterate:

  i. Include a latent variable and calculate probability of each LD block of containing a causal SNP per gene
  
  `\(w_{k} \sim \mbox{Ber}(\pi), \widetilde{y}_{k} \vert w_{k} = 1 \sim N( \widetilde{x}_{k}\beta, \sigma^2_{k})\)`

  ii. Fit weighted .important[Linear model]:
      
  `\(\widetilde{y}_{k} = \widetilde{x}_{k} \beta + \widetilde{\epsilon}_{k}\)`
        
  iii. Select another set of causal candidates 
]


---
class: top, left

## Description of the model for multiple phenotypes

.pull-left[

&lt;img src="figs/scatter_skew.png" width="450px" /&gt;

Genes: `\(g=1,\cdots,G\)`

LD blocks: `\(k = 1,\cdots, n_g\)`

SNPs: `\(i =1,\cdots,n_{gk}\)`

]

.pull-right[

Using a set of .important[regulatory] SNPs

1. Group the SNPs into LD blocks.

2. Given a set of causal candidates, iterate:

  i. Include a latent variable and calculate probability of each LD block of containing a causal SNP per gene
  
  `\(w_{gk} \sim \mbox{Ber}(\pi), \widetilde{y}_{gk} \vert w_{gk} = 1 \sim N( \widetilde{x}_{gk}\beta, \sigma^2_{gk})\)`

  ii. Fit weighted .important[L. Mixed model]:
      
  `\(\widetilde{y}_{gk} = \widetilde{x}_{gk} \beta + \widetilde{z}_{gk} u_g + \widetilde{\epsilon}_{gk}\)`
        
  iii. Select another set of causal candidates 
]

---
class: top, justify

# Advantages and future directions:

.pull-left[

### Advantages

1. Very flexible framework.

2. For a typical loci, an average run takes approx. 20 seconds. 

3. The method is fully integrated with the `lme4` R package. 
]

.pull-right[
### Future directions

1. Research strategies to pool several loci together.

2. Evaluate our model against state-of-the-art Fine-mapping methods (CAVIAR, PAINTOR, etc).

3. Include typical annotation features to the model as fixed-effects.
]

---
class: inverse, middle, center

# Acknowledgements

## Keles lab 

## Johannsen lab
    </textarea>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function() {
  var d = document, s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})();</script>

<script type="text/x-mathjax-config">
MathJax.Hub.Config({
  tex2jax: {
    skipTags: ['script', 'noscript', 'style', 'textarea', 'pre']
  }
});
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://cdn.bootcss.com/mathjax/2.7.1/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
