---
abstract: "ChIP-exo/nexus experiments rely on innovative modifications of the commonly used ChIP-seq protocol for high resolution mapping of transcription factor binding sites. Although many aspects of the ChIP-exo data analysis are similar to those of ChIP-seq, these high throughput experiments pose a number of unique quality control and analysis challenges. We develop a novel statistical quality control pipeline and accompanying R/Bioconductor package, ChIPexoQual, to enable exploration and analysis of ChIP-exo and related experiments. ChIPexoQual evaluates a number of key issues including strand imbalance, library complexity, and signal enrichment of data. Assessment of these features are facilitated through diagnostic plots and summary statistics computed over regions of the genome with varying levels of coverage. We evaluated our QC pipeline with both large collections of public ChIP-exo/nexus data and multiple, new ChIP-exo datasets from Escherichia coli. ChIPexoQual analysis of these datasets resulted in guidelines for using these QC metrics across a wide range of sequencing depths and provided further insights for modelling ChIP-exo data."
authors:
- admin
- Dongjun Chung
- Jeffrey Grass
- Robert Landick
- Sunduz Keles
date: "2017-07-29"
doi: "10.1093/nar/gkx594"
featured: true
image:
  caption: "Analysis of FoxA1 ChIP-exo in mouse liver"
  focal_point: "top"
  preview_only: false
projects: []
publication: 'Nucleic Acids Research 45(15) e145'
publication_short: ""
publication_types:
- "2"
publishDate: "2017-07-29"
slides: ""
summary: ""
title: "Data Exploration, quality control and statistical analysis of ChIP-exo/nexus experiments"
url_code: ""
url_dataset: ""
url_pdf: ""
url_poster: ""
url_project: ""
url_slides: ""
url_source: ""
url_video: ""
---
