---
date: "2015-07-03"
external_link: ""
image:
  caption: CTCF and Histone signal across CTCF peaks
  focal_point: center
share: false
links:
- icon: github
  icon_pack: fab
  link: "https://github.com/welch16/Segvis"
  name: Segvis
slides: ""
summary: An R package for ChIP-seq peaks visualization
title: Segvis 
url_code: ""
url_pdf: ""
url_slides: ""
url_video: ""
---

An R package to visualize ChIP-seq experiments across a set of pre-defined peaks. The package requires a set of pre-defined genomic regions, and aligned reads in bam format. We can do many tasks with it, some examples are:

* Extracts read data of specified input regions
* Plot peaks of different conditions across the same set of regions
* Plot central measures for a set of regions with the same width
* Subset regions according to used defined annotations
* Plot the heatmap of signal curves accross regions separated by annotation

The package is available [here](https://github.com/welch16/Segvis)
